tests/drivers/uart/uart_async_api/prj.conf

```diff
@@ -2,4 +2,4 @@ CONFIG_SERIAL=y
 CONFIG_UART_ASYNC_API=y
 CONFIG_UART_USE_RUNTIME_CONFIGURE=y
 CONFIG_ZTEST=y
-CONFIG_TEST_USERSPACE=y
+CONFIG_TEST_USERSPACE=n
```

Comment:
No. This is not correct. You should not disable usespace testing for the whole test on every platforms. Do it in board overlay just for the boards needed.